.github/workflows/ci.yaml

```diff
@@ -94,6 +94,30 @@ jobs:
           version: v1.55.2
           working-directory: ${{ matrix.directory }}
 
+  govulncheck:
+    name: govulncheck
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Setup go
+      - uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: govulncheck
```

Comment:
govulcheck shows 4 vulnerabilities on a local run that needs fixing for a CI run to pass for all PRs. Earlier run of these changes seemed to show no vulnerabilities, but once the CI starts working these may need to be addressed:

```
Scanning your code and 750 packages across 81 dependent modules for known vulnerabilities...

=== Symbol Results ===

Vulnerability #1: GO-2024-2610
    Errors returned from JSON marshaling may break template escaping in
    html/template
  More info: https://pkg.go.dev/vuln/GO-2024-2610
  Standard library
    Found in: html/template@go1.21.7
    Fixed in: html/template@go1.21.8
    Example traces found:
      #1: controllers/ramenconfig.go:240:18: controllers.NamespaceName calls os.Getenv, which eventually calls template.Template.Execute
      #2: controllers/ramenconfig.go:240:18: controllers.NamespaceName calls os.Getenv, which eventually calls template.Template.ExecuteTemplate

Vulnerability #2: GO-2024-2600
    Incorrect forwarding of sensitive headers and cookies on HTTP redirect in
    net/http
  More info: https://pkg.go.dev/vuln/GO-2024-2600
  Standard library
    Found in: net/http@go1.21.7
    Fixed in: net/http@go1.21.8
    Example traces found:
      #1: controllers/volsync/vshandler.go:1411:26: volsync.VSHandler.GetVolumeSnapshotClasses calls client.client.List, which eventually calls http.Client.Do
      #2: controllers/s3utils.go:520:51: controllers.s3ObjectStore.ListKeys calls s3.S3.ListObjectsV2WithContext, which eventually calls http.Client.Get

Vulnerability #3: GO-2024-2599
    Memory exhaustion in multipart form parsing in net/textproto and net/http
  More info: https://pkg.go.dev/vuln/GO-2024-2599
  Standard library
    Found in: net/textproto@go1.21.7
    Fixed in: net/textproto@go1.21.8
    Example traces found:
      #1: controllers/s3utils.go:520:51: controllers.s3ObjectStore.ListKeys calls s3.S3.ListObjectsV2WithContext, which eventually calls textproto.Reader.ReadLine
      #2: controllers/s3utils.go:583:44: controllers.s3ObjectStore.DownloadObject calls json.Decoder.Decode, which eventually calls textproto.Reader.ReadMIMEHeader

Vulnerability #4: GO-2024-2598
    Verify panics on certificates with an unknown public key algorithm in
    crypto/x509
  More info: https://pkg.go.dev/vuln/GO-2024-2598
  Standard library
    Found in: crypto/x509@go1.21.7
    Fixed in: crypto/x509@go1.21.8
    Example traces found:
      #1: controllers/s3utils.go:583:44: controllers.s3ObjectStore.DownloadObject calls json.Decoder.Decode, which eventually calls x509.Certificate.Verify

Your code is affected by 4 vulnerabilities from the Go standard library.
This scan also found 1 vulnerability in packages you import and 2
vulnerabilities in modules you require, but your code doesn't appear to call
these vulnerabilities.
Use '-show verbose' for more details.
```